lib/cfn-nag/custom_rules/LambdaFunctionCloudWatchLogsRule.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+require 'cfn-nag/violation'
+require_relative 'base'
+
+class LambdaFunctionCloudWatchLogsRule < BaseRule
+  def rule_text
+    'Lambda functions require permission to write CloudWatch Logs'
+  end
+
+  def rule_type
+    Violation::WARNING
+  end
+
+  def rule_id
+    'W58'
+  end
+
+  def audit_impl(cfn_model)
+    # Iterate over each Lambda function
+    lambda_functions = cfn_model.resources_by_type('AWS::Lambda::Function')
+    violating_lambda_functions = lambda_functions.select do |lambda_function|
+      # Throw warning if no associated role object
+      next lambda_function if lambda_function.role_object.nil?
+
+      # Add lambda as violating if meets conditions
+      violating_role?(lambda_function.role_object)
+    end
+
+    violating_lambda_functions.map(&:logical_resource_id)
+  end
+
+  def managed_policies_include_cw_logs_access?(managed_policies)
+    managed_policies.include?('arn:aws:iam::aws:policy/CloudWatchLogsFullAccess') || \
```

Comment:
nit: one thing to consider is using intersection() instead.  that way you can corral the policies names in one array

Reply:
Looks like `intersection()` wasn't available in this version of ruby, but I was able to gain the same functionality using the `&` operator in the latest commit.